doggos/knowledge/rule.py

```diff
@@ -1,24 +1,105 @@
-from typing import Dict, Tuple, List
+from typing import Dict, NoReturn, List, NewType
 
 from doggos.fuzzy_sets.membership.membership_degree import MembershipDegree
-from doggos.knowledge.antecedent import Antecedent
+from doggos.knowledge import Antecedent, Clause
```

Comment:
Change it to two separate imports with full paths, because it might cause problems.